### PR TITLE
Implement skeleton for Test Details Views

### PR DESCRIPTION
**Pull request by sarathms**
Implements 1 out of 7 views listed in #3

### CODE REVIEW

**Comment by sarathms** [comment]
As of now, we have a working structure. Specific data like the content for Hero (and an alternative collapsed Hero), test details come from its own `/nettests/websites/WebConnectivity.js` file. This file also holds metadata about that test.
Common content will be generated and rendered in `MeasurementContainer` itself.

Also [added a collection](https://github.com/ooni/probe-desktop/pull/44/files#diff-4bf25f7c89f68f5d68796f5670e71217R94-R111) called `tests` in `nettestes/index.js` to map metadata for all test. So far we only export the translated name of the test. More can be added.

@hellais Please review and point out holes in this approach.

image


image

**Comment by hellais** [comment]
> As of now, we have a working structure. Specific data like the content for Hero (and an alternative collapsed Hero), test details come from its own /nettests/websites/WebConnectivity.js file. This file also holds metadata about that test.
> Common content will be generated and rendered in MeasurementContainer itself.

> Also added a collection called tests in nettestes/index.js to map metadata for all test. So far we > only export the translated name of the test. More can be added.

I took a look at the code as well and this approach looks solid to me 👍

**Comment by hellais** [comment]
Is it useful if I make a mockup of how the views in collapsed vs expanded form shall look like?

**Comment by bassosimone** [comment]
I suggest to merge this PR and then follow up with others! :shipit:

**Comment by sarathms** [comment]
I had made some changes to the approved model because I immediately found a problem with it. I will send them for review here in the next few minutes.

**Comment by sarathms** [comment]
The latest commit helps  compose a few common UI elements in the Hero (like Test Name, Date, Network) in the Container component, instead of all the test specific components. This also prevents having to pass in the `<BackButton>` component deeper into the hierarchy.

image

File: renderer/components/nettests/index.js [hunk 1]
```diff
@@ -89,6 +91,24 @@ export const testGroups = {
   }
 }
 
+export const tests = {
+  web_connectivity,
```
[the remark this answers is not in the document]
Comment (sarathms) [reply renderer/components/nettests/index.js]: That's correct.

File: renderer/components/nettests/websites/WebConnectivity.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          <Box>  </Box>
+        </Flex>
+      </Box>
+    </Flex>
+  )
+
+  return (
+    <div>
+      {render({
+        heroBG: 'green8', // TODO: Determine color based on presence of anomaly
```
[the remark this answers is not in the document]
Comment (sarathms) [reply renderer/components/nettests/websites/WebConnectivity.js]: Added

File: renderer/components/nettests/index.js [hunk 3]
```diff
@@ -89,6 +91,24 @@ export const testGroups = {
...
+  http_header_field_manipulation: web_connectivity,
+  http_invalid_request_line: web_connectivity,
+  telegram: web_connectivity,
+  whatsapp: web_connectivity,
+  ndt: web_connectivity,
+  dash: web_connectivity,
+  vanilla_tor: web_connectivity,
```
[the remark this answers is not in the document]
Comment (sarathms) [reply renderer/components/nettests/index.js]: Added TODO

File: renderer/components/measurement/MeasurementContainer.js [hunk 4]
```diff
@@ -97,32 +99,144 @@ const HeaderContent = styled(Box)`
   -webkit-app-region: drag;
 `
 
+const detailsMap = {
+  web_connectivity: WebConnectivity,
+  facebook_messenger: FacebookMessengerDetails
```
[the remark this answers is not in the document]
Comment (sarathms) [reply renderer/components/measurement/MeasurementContainer.js]: Added

**Pull request merged**